cEP-0014.md

```diff
@@ -0,0 +1,353 @@
+    2. Among the remaining bears in `candidate_bears`, if some `ProjectDependenyInfo` extracted from the project matches the `REQUIREMENTS` field of bear (matching might also involve checking the compatibility of semvers), move the tuple `(Info, Bear)` to  `to_propose_bears`.
+
+7. For every bear in `to_propose_bears`:
+    1. If there's enough extracted information from the project to fill in all the non-optional settings of the Bear, move the Bear to `selected_bears` list.
+    2. Else, prompt the user on CLI with the evidence (the source of  `Info` instance in the tuple) to enable the bear. If the user agrees, move the bear to `selected_bears`, otherwise discard the bear.
+
+8. For each of the remaining bears in `candidate_bears`, eliminate bears having capabilities already satisfied in `selected_bears`.
+
+9. Among the remaining bears in `candidate_bears`, in the case of conflicts in capabilities among the bears in the list:
+    1. Eliminate bears having no unique capabilities among the other bears present in the list.
+    2. Give preference to the bears already having dependencies installed (using the `check_prerequisites` field of the bears) and eliminate the respective conflicting bears.
+    3. Add the remaining bears to `selected_bears` list.
+
+10. For every non-optional setting of each bear `selected_bears` object,
+    1. If the setting's value exists in extracted information, do nothing.
+    2. Else,
+        1. If the quickstart is being executed in interactive mode, prompt the user for the setting value.
+        2. If non-interactive mode, discard the bear.
+
+11. Finally, create sections from the `selected_bears` list using the collected information values from user inputs and project files.
+
+In future, more constraints can be added to further narrow down the list of selected bears by giving users more flexibility in terms of options like `max_bears`, `max_bears_per_language`, `exclude_languages`, etc.
+
+
+### Mapping extracted information to setting values.
+
+To automatically fill in some of the setting values in .coafile sections, the `Info` classes need to be mapped to settings values of the Bears. These scope of applicability of these `Info` instances may vary from "bear-specific", to "section" to "global".
+The scope for a mapping may be defined based on the:
+- Source of the `Info` instance
+- Value of the `Info` instance
+- Class of the `Info` instance
+- Any other attribute encapsulated in the `Info` instance
+
+The scope of the information can be represented by the `InfoScope` class as follows:
+```py
+class InfoScope:
+    def __init__(self,
+                 level,
+                 sections=[],
+                 bears=[],
+                 allowed_sources=[],
+                 allowed_extractors=[]):
+        """
+        A class representing scope of applicability of an
+        ``Info`` instance.
+        :param level:              Broad-level scope, possible values are
```

Comment:
I think we need a newline before the params (I know it's just a cEP, but still :P)